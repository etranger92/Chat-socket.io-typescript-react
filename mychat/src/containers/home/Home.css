@import "../../style/shared.css";

#home {
    margin: 2rem;
}

#container-select-room {
    width: 20vw;
    display: flex;
    flex-direction: column;
    justify-content: center;

}

#container-select-room button {
    width: 50%;
    align-self: center;
    margin-top: 2rem;

}