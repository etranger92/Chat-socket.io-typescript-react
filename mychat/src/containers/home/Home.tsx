import React from 'react';
import { Component } from 'react';
import './Home.css';
import { Chat } from '../../components/chat/Chat';
// Using class => import Chaton from '../../components/chat/Chaton';

interface IState {
  room: string;
  isRoomPicked: boolean;
}
interface IProps {}

class Home extends Component<IProps, IState> {
  private inputRef: React.RefObject<HTMLSelectElement>;
  constructor(props: any) {
    super(props);
    this.state = {
      room: '',
      isRoomPicked: false,
    };
    this.inputRef = React.createRef();
  }

  handleClick = () => {
    let value = this.inputRef.current?.value;
    this.setState({
      room: value!,
      isRoomPicked: true,
    });
    //console.log(this.state.room, 'your room when clicked');
  };

  render() {
    const { room } = this.state;
    return (
      <>
        <div id='home'>
          <h1> Hello home</h1>
          <div id='container-select-room'>
            <select name='rooms' id='rooms' ref={this.inputRef}>
              <option value='Sport'>Sport</option>
              <option value='Science'>Science</option>
            </select>
            <button onClick={this.handleClick}> Join</button>
          </div>
        </div>
        {/*  Using Class {this.state.isRoomPicked && <Chaton roomName={room} />} */}
        {/* Using functional component */}{' '}
        {this.state.isRoomPicked && <Chat roomName={room} />}
      </>
    );
  }
}

export default Home;
