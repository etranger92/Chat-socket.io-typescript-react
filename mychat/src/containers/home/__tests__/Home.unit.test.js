import React from 'react';
import { shallow } from 'enzyme';
import Home from '../Home';

describe('We test the innerHTML of the component', () => {
  test('renders Home', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
