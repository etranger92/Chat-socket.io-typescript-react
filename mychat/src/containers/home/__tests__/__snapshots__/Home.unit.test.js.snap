// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We test the innerHTML of the component renders Home 1`] = `
<Fragment>
  <div
    id="home"
  >
    <h1>
       Hello home
    </h1>
    <div
      id="container-select-room"
    >
      <select
        id="rooms"
        name="rooms"
      >
        <option
          value="Sport"
        >
          Sport
        </option>
        <option
          value="Science"
        >
          Science
        </option>
      </select>
      <button
        onClick={[Function]}
      >
         Join
      </button>
    </div>
  </div>
   
</Fragment>
`;

exports[`renders HeaderNav 1`] = `
<Fragment>
  <div
    id="home"
  >
    <h1>
       Hello home
    </h1>
    <div
      id="container-select-room"
    >
      <select
        id="rooms"
        name="rooms"
      >
        <option
          value="Sport"
        >
          Sport
        </option>
        <option
          value="Science"
        >
          Science
        </option>
      </select>
      <button
        onClick={[Function]}
      >
         Join
      </button>
    </div>
  </div>
   
</Fragment>
`;

exports[`renders Home 1`] = `
<Fragment>
  <div
    id="home"
  >
    <h1>
       Hello home
    </h1>
    <div
      id="container-select-room"
    >
      <select
        id="rooms"
        name="rooms"
      >
        <option
          value="Sport"
        >
          Sport
        </option>
        <option
          value="Science"
        >
          Science
        </option>
      </select>
      <button
        onClick={[Function]}
      >
         Join
      </button>
    </div>
  </div>
   
</Fragment>
`;
